### ISSUE TITLE
Improved templating

### ISSUE THREAD

**Issue by pelson** [conda-forge/conda-forge.github.io#5]
Conda smithy has a hardcoded number of files in https://github.com/conda-forge/conda-smithy/blob/master/conda_smithy/configure_feedstock.py which are generated into the resulting feedstock. Remove the bespoke functions, and just generate all files in the https://github.com/conda-forge/conda-smithy/tree/master/templates directory.

For feedstocks which need to override the templates, allow a templates directory in the feedstock which can inherit from conda-smithy (e.g. ```{% extends "conda-smithy/README.tmpl" %}```) for truly bespoke customisation.

**Comment by JanSchulz** [comment]
Shouldn't this go to the conda smithy repo?

**Comment by pelson** [comment]
> Shouldn't this go to the conda smithy repo?

Yes, it was an attempt to kindle some interest by pulling together the required work into a single issue tracker. Canonically, this issue is a conda-smithy issue. As you notice, the complexity is low, and the size is medium, yet nobody has been forthcoming with a contribution. If you're interested, I'd happily accept a *standalone* :wink: PR which addresses this.
